### ISSUE TITLE
Examples (lots of) with a broken sense of FP

### ISSUE THREAD

**Issue by ac2epsilon** [scala/docs.scala-lang#2315]
```
i match
  case 0 => println("1")
  case 1 => println("2")
  case what => println(s"You gave me: $what" )
```
While things like this just work, far better from the beginning insulate pure (adt) code from impure (smelling) one. Sure the following interpretation is more Scal-ish than the former, simplifies refactoring, and after all saves typing:
```
val r = i match
  case 0 => "1"
  case 1 => "2"
  case what => s"You gave me: $what"
println(r)
```

**Comment by julienrf** [comment]
That’s a good point, thank you for creating this issue. Would you be interested in submitting a fix?

**Comment by ac2epsilon** [comment]
Good, it's mine, then.

Not a fix actually, it is rather some code review.

Codename un-print-l-n-isation(-itation?). I will use it to mark all related
PRs. Any suggestions?

To hold control I limit the scope to Scala 3 book.

The plan is the following: open files in order, look for println,
paraphrase code to remove println-s (but one), try in Scastie, commit in
loop, push, PR in loop.

Looks like a simple bot's job.

I expect to spend some days/couple of weeks on this in relaxed mode. To
avoid the all-or-nothing case I will pack some commits/files processed into
single PR and so forth.

Does it all of this relevant and adequate?

Also, I sent some other ideas, but as they say, let's one thread at a time.

**Comment by julienrf** [comment]
I don't think we should systematically remove all the println statements. In the example you gave, that would improve the readability (although I would rename the val to 'message' instead of 'r'), with the added benefit of showing a purely functional match expression.

**Comment by ac2epsilon** [comment]
Got it. Working.

вт, 8 лют. 2022 р. о 09:04 Julien Richard-Foy ***@***.***>
пише:
